generator/template.go

```diff
@@ -195,6 +195,97 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+type RequestPayload = Record<string, unknown>;
+type Primitive = string | boolean | number;
+
+/**
+ * Checks if given value is of a primitive type
+ * @param  {unknown} value
+ * @return {boolean}
+ */
+function isPrimitive(value: unknown): boolean {
+  return (
+    typeof value === "string" ||
+    typeof value === "number" ||
+    typeof value === "boolean"
+  );
+}
+
+/**
+ * Flattens a deeply nested request payload and returns an object
+ * with only primitive values and non-empty array of primitive values
+ * @param  {RequestPayload} requestPayload
+ * @param  {String} path
+ * @return {Record<string, Primitive | Array<Primitive>}
+ */
+function flattenRequestPayload<T extends RequestPayload>(
+  requestPayload: T,
+  path: string = ""
+): Record<string, Primitive | Array<Primitive>> {
+  return Object.keys(requestPayload).reduce(
+    (acc: T, key: string): T => {
+      const value = requestPayload[key];
+      const newPath = path ? [path, key].join(".") : key;
+
+      const isObject =
+        Object.prototype.toString.call(value) === "[object Object]";
+      const isNonEmptyPrimitiveArray =
+        Array.isArray(value) &&
+        value.every(v => isPrimitive(v)) &&
+        value.length > 0;
+
+      switch (true) {
```

Comment:
`switch(true)` seems quite odd to me. you could turn this block into something like the following

```
let objectToMerge = {}

if (isObject) {
  objectToMerge = flattenRequestPayload(value as RequestPayload, newPath)
} else if (isPrimitive(value) || isNonEmptyPrimitiveArray) {
  objectToMerge = {[newPath]: value}
}

return {...acc, ...objectToMerge}
```

Reply:
Thank you for your feedback. I will revise this as per your suggestion 🙇🏼